### PR TITLE
fix: filter out inventory events with nil UUID insights_id

### PR DESCRIPTION
**Pull request by jpramos123**
Replicate the insightsOnly Kafka Connect filter in the inventory consumer: reject hosts where insights_id equals the nil UUID (00000000-0000-0000-0000-000000000000) in addition to the existing null/empty check. Adds a dedicated prometheus counter for observability.

## Summary by Sourcery

Filter out inventory events where insights_id is the nil UUID and track them via a dedicated Prometheus counter.

New Features:
- Introduce a Prometheus counter for inventory events rejected by the insights-only filter.

Bug Fixes:
- Prevent hosts with a nil UUID insights_id from being ingested in the inventory replication flow.

Enhancements:
- Add explicit constant for the nil UUID and improve logging when inventory events are rejected due to invalid insights_id.

Tests:
- Add a test to verify that inventory events with a nil UUID insights_id are filtered out and not upserted.

### CODE REVIEW

File: api/advisor/api/management/commands/advisor_inventory_service.py [hunk 1]
```diff
@@ -264,12 +266,13 @@ def parse_created_event(message: dict[str, JsonValue]) -> ParsedInventoryHost |
...
-    if not insights_id:
+    if not insights_id or insights_id == NIL_UUID:
         logger.error(
-            "Request %s: Inventory %s event has null or empty insights_id for host %s",
+            "Request %s: Inventory %s event has null, empty, or nil insights_id for host %s",
             request_id, event_type, host_id
         )
         prometheus.INVENTORY_EVENT_MISSING_KEYS.inc()
+        prometheus.INVENTORY_EVENT_INSIGHTS_ONLY_FILTERED.inc()
         return None
```
Comment (rodrigonull) [inline api/advisor/api/management/commands/advisor_inventory_service.py]: I would change this to:


```suggestion
if not insights_id:
    # Defensive: HBI always provides insights_id, but guard against upstream changes
    logger.warning(
        "Request %s: Inventory %s event has null or empty insights_id for host %s",
        request_id, event_type, host_id
    )
    prometheus.INVENTORY_EVENT_MISSING_KEYS.inc()
    return None
if insights_id == NIL_UUID:
    logger.info(
        "Request %s: Inventory %s event filtered — host %s has nil insights_id",
        request_id, event_type, host_id
    )
    prometheus.INVENTORY_EVENT_INSIGHTS_ONLY_FILTERED.inc()
    return None
```

Comment (jpramos123) [reply api/advisor/api/management/commands/advisor_inventory_service.py]: I like it

File: api/advisor/prometheus.py [hunk 2]
```diff
@@ -27,6 +27,10 @@
     'insights_advisor_api_inventory_event_malformed',
     'Counter for inventory events that failed validation or parsing'
 )
+INVENTORY_EVENT_INSIGHTS_ONLY_FILTERED = Counter(
+    'insights_advisor_api_inventory_event_insights_only_filtered',
+    'Counter for inventory events filtered out due to missing or nil insights_id'
```
Comment (rodrigonull) [inline api/advisor/prometheus.py]:
```suggestion
    'Counter for inventory events filtered out due to nil insights_id (insightsOnly filter)'
```

**Review by rodrigonull — APPROVED** [review]
